components/builder-db/src/models/projects.rs

```diff
@@ -0,0 +1,111 @@
+use super::db_id_format;
+use chrono::NaiveDateTime;
+use diesel;
+use diesel::pg::PgConnection;
+use diesel::result::QueryResult;
+use diesel::sql_types::{BigInt, Bool, Text};
+use diesel::RunQueryDsl;
+use schema::project::*;
+
+#[derive(Debug, Serialize, Deserialize, QueryableByName)]
+#[table_name = "origin_projects"]
+pub struct Project {
+    #[serde(with = "db_id_format")]
+    pub id: i64,
+    #[serde(with = "db_id_format")]
+    pub origin_id: i64,
+    #[serde(with = "db_id_format")]
+    pub owner_id: i64,
+    pub origin_name: Option<String>,
+    pub package_name: Option<String>,
+    pub name: Option<String>,
+    pub plan_path: Option<String>,
```

Comment:
I see in the create/update structs below a lot of the things that are optional here are required there. Does it make sense to convert these to not use `Options`?

Reply:
Yeah that would be better if it works - I've only wired these as Options to match the DB columns.